scipy/special/orthogonal.pyi

```diff
@@ -285,8 +285,7 @@ class orthopoly1d(numpy.poly1d):
     @property
     def limits(self) -> Tuple[float, float]: ...
     def weight_func(self, x: float) -> float: ...
-    # TODO: ArrayLike
-    def __call__(self, x: Any) -> Any: ...
+    def __call__(self, x: numpy.typing.ArrayLike) -> numpy.typing.ArrayLike: ...
```

Comment:
The return should not be `ArrayLike` here I think. `ArrayLike` is a `Union` which includes a lot of types. A union is almost never the correct thing to return. In this case, `np.poly1d` doesn't have annotations yet, but it looks like `__call__` can return either a `float` or an `ndarray`. Would be good to verify that in the test. Then if there are clear conditions for when it returns one or the other, you can use `@overload`. Otherwise I think it's best to leave the return annotation as `Any`.